src/shared/codelens/localLambdaRunner.ts

```diff
@@ -310,3 +330,320 @@ export class LocalLambdaRunner {
+    ).replace('\\', '/')
+
+    // tslint:disable-next-line:max-line-length
+    const workspaceFolder = vscode.workspace.getWorkspaceFolder(params.workspaceUri) // localInvokeParams.workspaceFolder.uri
+    let existingTemplateResource: CloudFormation.Resource | undefined
+    if (workspaceFolder) {
+        const lambdas = await detectLocalLambdas([workspaceFolder])
+        const existingLambda = lambdas.find(lambda => lambda.handler === relativeFunctionHandler)
+        existingTemplateResource = existingLambda ? existingLambda.resource : undefined
+    }
+
+    let newTemplate = new SamTemplateGenerator()
+        .withCodeUri(params.codeDir)
+        .withFunctionHandler(relativeFunctionHandler)
+        .withResourceName(TEMPLATE_RESOURCE_NAME)
+        .withRuntime(params.runtime)
+
+    if (existingTemplateResource && existingTemplateResource.Properties &&
+        existingTemplateResource.Properties.Environment) {
+        newTemplate = newTemplate.withEnvironment(existingTemplateResource.Properties.Environment)
+    }
+
+    await newTemplate.generate(inputTemplatePath)
+
+    return inputTemplatePath
+}
+
+export async function executeSamBuild(params: {
+    baseBuildDir: string,
+    channelLogger: ChannelLogger,
+    codeDir: string,
+    inputTemplatePath: string,
+    manifestPath?: string,
+    samProcessInvoker: SamCliProcessInvoker,
+}): Promise<string> {
+    params.channelLogger.info(
+        'AWS.output.building.sam.application',
+        'Building SAM Application...'
+    )
+
+    const samBuildOutputFolder = path.join(params.baseBuildDir, 'output')
+
+    const samCliArgs: SamCliBuildInvocationArguments = {
+        buildDir: samBuildOutputFolder,
+        baseDir: params.codeDir,
+        templatePath: params.inputTemplatePath,
+        invoker: params.samProcessInvoker
+    }
+    if (params.manifestPath) {
+        samCliArgs.manifestPath = params.manifestPath
+    }
+    await new SamCliBuildInvocation(samCliArgs).execute()
+
+    params.channelLogger.info(
+        'AWS.output.building.sam.application.complete',
+        'Build complete.'
+    )
+
+    return path.join(samBuildOutputFolder, 'template.yaml')
+}
+
+export async function invokeLambdaFunction(params: {
+    baseBuildDir: string,
+    channelLogger: ChannelLogger,
+    configuration: SettingsConfiguration,
+    debugConfig: DebugConfiguration,
+    documentUri: vscode.Uri,
+    handlerName: string,
+    isDebug?: boolean,
+    samTemplatePath: string,
+    samTaskInvoker: SamCliTaskInvoker,
+    onWillAttachDebugger?(): Promise<void>,
+}): Promise<void> {
+    params.channelLogger.info(
+        'AWS.output.starting.sam.app.locally',
+        'Starting the SAM Application locally (see Terminal for output)'
+    )
+    params.channelLogger.logger.info(`localLambdaRunner.invokeLambdaFunction: ${JSON.stringify(
+        {
+            baseBuildDir: params.baseBuildDir,
+            // configuration: SettingsConfiguration, // Can we pretty print this?
```

Comment:
Any reason to not (un-pretty) print this in the meanwhile?

Reply:
Checked. It prints fine. Fixed.
✔️